release

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env -S bash -e
+
+cd $(dirname $0)
+
+main() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  [[ -z "$version" ]]      && usage "Missing version number"
+  [[ -z "$preview_txid" ]] && usage "Missing preview network hydra scripts publication transaction id"
+  [[ -z "$preprod_txid" ]] && usage "Missing preprod network hydra scripts publication transaction id"
+
+  check_can_release "$version"
+
+  prepare_release "$version" "$preview_txid" "$preprod_txid"
+
+  publish_release # fake for now
+}
+
+usage() {
+  local message=$1
+
+  cat >&2 <<EOF
+$message
+
+  $0 <version> <preview_txid> <preprod_txid>
+Publishes a new release of hydra.
+<version> must respect [Semantic Versioning](http://semver.org/)
+<preview_txid> is the id of the transaction used to publish the hydra scripts on the preview network (see smoke tests)
+<preprod_txid> is the id of the transaction used to publish the hydra scripts on the preprod network (see smoke tests)
+EOF
+
+  exit 1
+}
+
+check_can_release() {
+  local version="$1"
+
+  check_no_uncommited_changes  
+  check_version_is_valid $version
+  check_changelog_is_up_to_date $version
+
+  true #avoid error on last instruction of function (see bash -e)
+}
+
+prepare_release() {
+  local version="$1"
+  local preview_txid="$2"
+  local preprod_txid="$3"
+
+  update_cabal_version "$version"
+  update_api_version   "$version"
+  update_preview_txid "$version" "$preview_txid"
+  update_preprod_txid "$version" "$preprod_txid"
+
+  find . -name '*-e' -exec rm '{}' \; # cleanup BSD sed mess
+  
+  git add .
+  
+  git commit -m "prepare release $version"
+ 
+  git tag -as "$version" -F <(changelog "$version")
```

Comment:
Could use directly stdin. IIRC that would be `git tag -as "$version" -F - <<< changelog "$version"`, but I'm not sure.

Reply:
I think it would more look like this as `<<<` is for static strings if I'm correct (but I'm not use to this `<<<` thing so I might be wrong):

```bash
#> changelog "$version" | git tag -as "$version" -F -
```

Reply:
Unless it's a problem, I'll keep with the pseudo file process substitution version. I feel there is less movement to read the expression. But I guess it's a matter of taste.